### ISSUE TITLE
Option to run async tests in pararell on a single process

### ISSUE THREAD

**Issue by kuba-orlik** [mochajs/mocha#4537]
**Is your feature request related to a problem or a nice-to-have?? Please describe.**

Running tests in parallel with mocha adds the overhead of using and spawning separate workers, which makes it slower than it has to be in test suites based on a lot of I/O and async behavior.

**Describe the solution you'd like**
Option to run all the tests at once in a single worker. Think of a `Promise.all ` for all the `it`s

**Describe alternatives you've considered**
Not sure how to achieve that effect in any other way

**Comment by nopeless** [comment]
#4588 I made an issue similar to this but with detailed plans @kuba-orlik

**Comment by JoshuaKGoldberg** [comment]
Same as https://github.com/mochajs/mocha/issues/4588#issuecomment-1870704969, alas.
